plugins/single_plugins/oswitch.cpp

```diff
@@ -45,6 +46,102 @@ class wayfire_output_manager : public wf::plugin_interface_t
         return true;
     };
 
+
+    wf::wl_idle_call idle_previous_output;
+
+    wf::activator_callback switch_output_previous = [=] (auto)
```

Comment:
In gneneral, I feel like quite a lot of code is duplicated for both directions. It would be best to add helper functions to avoid this duplication.